src/features/dish_details/Dish.tsx

```diff
@@ -3,84 +3,90 @@ import {View, TouchableOpacity, Image, Dimensions, ScaledSize, Text, ScrollView}
-  const scroll = useRef<Animated.ScrollView>(null);
-  const {scrollHandler, x} = useScrollHandler();
+export const Dish = inject(Stores.DishStore)(
+  observer(({navigation, route, dish}: Props) => {
+    const scroll = useRef<Animated.ScrollView>(null);
+    const {scrollHandler, x} = useScrollHandler();
 
-  return (
-    <ScrollView style={styles.container}>
-      <View style={styles.header}>
-        <TouchableOpacity style={styles.backButton} onPress={navigation.goBack}>
-          <BackIcon name="arrow-back-ios" size={24} color="#000000" />
-        </TouchableOpacity>
-        <TouchableOpacity style={styles.backButton}>
-          <FavouriteIcon name="heart" size={20} color="#000000" />
-        </TouchableOpacity>
-      </View>
-      <View>
-        <Animated.ScrollView
-          ref={scroll}
-          horizontal
-          contentContainerStyle={styles.imageBlock}
-          snapToInterval={width}
-          bounces={false}
-          style={{width}}
-          showsHorizontalScrollIndicator={false}
-          {...scrollHandler}>
-          <View style={styles.imageItem}>
-            <Image source={{uri: route.params.dish.image}} style={styles.image} />
-          </View>
-          <View style={styles.imageItem}>
-            <Image source={{uri: route.params.dish.image}} style={styles.image} />
-          </View>
-          <View style={styles.imageItem}>
-            <Image source={{uri: route.params.dish.image}} style={styles.image} />
-          </View>
-          <View style={styles.imageItem}>
-            <Image source={{uri: route.params.dish.image}} style={styles.image} />
-          </View>
-        </Animated.ScrollView>
-        <Animated.View style={styles.dotItem}>
-          <Dot index={0} currentIndex={divide(x, width)} />
-          <Dot index={1} currentIndex={divide(x, width)} />
-          <Dot index={2} currentIndex={divide(x, width)} />
-          <Dot index={3} currentIndex={divide(x, width)} />
-        </Animated.View>
-      </View>
-      <View style={styles.titleBlock}>
-        <Text style={styles.title}>{route.params.dish.name}</Text>
-        <Text style={styles.cost}>N{route.params.dish.cost}</Text>
-      </View>
-      <View style={styles.description}>
-        <View style={styles.infoBlock}>
-          <Text style={styles.nameInfo}>Delivery info</Text>
-          <Text style={styles.info}>{route.params.dish.info}</Text>
+    return (
+      <ScrollView style={styles.container}>
+        <View style={styles.header}>
+          <TouchableOpacity style={styles.backButton} onPress={navigation.goBack}>
+            <BackIcon name="arrow-back-ios" size={24} color="#000000" />
+          </TouchableOpacity>
+          <TouchableOpacity onPress={() => dish.addFavourite(route.params.dish.id)} style={styles.backButton}>
+            <MaterialCommunityIcons name={route.params.dish.favourite ? 'heart' : 'heart-outline'} size={22} color="red" />
+          </TouchableOpacity>
```

Comment:
@DiamondRock597 we can create special Button to not do this twice directly in screen component.